storage/StorageInterface.h

```diff
@@ -214,6 +214,23 @@ namespace storage
 
 
     /**
+     * Abstract class for some callbacks during commit. Currently only works
+     * for btrfs as root filesystem.
+     */
+    class CommitCallbacks
+    {
+    public:
+
+	virtual ~CommitCallbacks() {}
+
+	virtual void post_root_filesystem_create() const {}
+	virtual void post_root_mount() const {}
+	virtual void post_root_fstab_add() const {}
```

Comment:
That would require users to override all functions instead of just those they nerd. This is esp. bad if functions are added since all users must be updated.